runtime/karura/src/lib.rs

```diff
@@ -1340,6 +1342,19 @@ parameter_types! {
 
 pub type Barrier = (TakeWeightCredit, AllowTopLevelPaidExecutionFrom<All<MultiLocation>>);
 
+pub struct ToTreasury;
+impl TakeRevenue for ToTreasury {
+	fn take_revenue(revenue: MultiAsset) {
+		if let MultiAsset::ConcreteFungible { id, amount } = revenue {
+			if CurrencyIdConvert::convert(id).is_some() {
```

Comment:
`is_some` check only ensures `id` could be converted to a `CurrencyId` but it doesn't have to be KSM.

To handle it correctly, get the wrapped `currency_id` in result and deposit with it.